addons/mail/static/tests/owl/component/chat_window_manager_tests.js

```diff
@@ -601,6 +602,281 @@ QUnit.test('chat window: open / close', async function (assert) {
+        await testUtils.nextTick(); // re-render
+    }
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                is_minimized: false,
+                name: "General",
+                state: 'open',
+                uuid: 'channel-20-uuid',
+            }],
+        },
+    });
+    const channelId = 20;
+    const localChannelId = `mail.channel_${channelId}`;
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [{
+                    id: 20,
+                    last_message: {
+                        author_id: [7, "Demo"],
+                        body: "<p>test</p>",
+                        channel_ids: [20],
+                        id: 100,
+                        message_type: 'comment',
+                        model: 'mail.channel',
+                        res_id: 20,
+                    },
+                }];
+            } else if (args.method === 'channel_fold') {
+                return [];
+            } else if (args.method === 'message_fetch'){
+                const out = [];
+                for (let i = 0; i < 20; i++) {
+                    out.push({
+                        author_id: [11, "Demo"],
+                        body: "<p>body</p>",
+                        channel_ids: [20],
+                        date: "2019-04-20 10:00:00",
+                        id: i+10,
+                        message_type: 'comment',
+                        model: 'mail.channel',
+                        record_name: 'General',
+                        res_id: i+1,
+                    });
+                }
+                return out;
+            }
+            return this._super(...arguments);
+        },
+    });
+    window.openThread = openThread;
+    await openThread();
+
+    const scrollTop = 142;
+    const composerHTML = "XDU for the win !";
+    const composerAttachments = ['a.png', 'b.pdf'];
+
+    // SETUP INITIAL STATE
+    // - Set a scroll position to chat window
+    document.querySelector(`.o_MessageList.o_Thread_messageList`).scrollTop = scrollTop;
+
+    // - Set html content of the composer of the chat window
+    const composerTextInput = document.querySelector(`
+        .o_ComposerTextInput
+        > .note-editor
+        > .note-editing-area
+        > .note-editable`);
+    composerTextInput.innerHTML = composerHTML;
+    composerTextInput.dispatchEvent(new window.KeyboardEvent('input'));
+    await testUtils.nextTick();
+
+    // TESTS
+
+    // - phase 1 : on hiding menu
+    // -- Hide home menu
+    await this.widget.call('chat_window', 'test:will_hide_home_menu');
+    await this.widget.call('chat_window', 'test:hide_home_menu');
+    let chatWindowStates;
+
+    // -- Check store state
+    chatWindowStates = this.store.state.chatWindowManager.storedChatWindowStates;
```

Comment:
Okay so basically I just need to split the test in two ? Moving all store checks to a new messaging_actions_tests ?

Reply:
1. In this test, no store state checks: only assertions of UI.
2. In messaging action tests, check this action makes correct store state changes.
3. Additional chat window manager test to check store updates stored chat window state on show/hide home menu.

Test 1. checks chat window manager behaves correctly
Test 2. checks store behaves correctly in isolation
Test 3. checks store behaves correctly from chat window manager actions

Test 1 is the most important one, but it hides some implementation details.
That's why 2 and 3 are useful.

Reply:
I understand the two first tests but I'm not sure if the test 3 is needed ?
The only action used is updateChatWindowManager which basically is just an `Object.assign` ... So it would mean we test the langage itself... what do you think ?

Reply:
Ok, can omit this one

Reply:
Okay :D